commit 959bbe4636373bd4e9a18ac470476803cc2087d0 (refs/changes/58/2458/2)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-08-24T22:55:39+00:00 (2 years, 1 month ago)
    
    OS-6308 fix for OS-6275 blows assert on debug kernel
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
index cdada58f15..b96727cd8a 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
@@ -186,7 +186,9 @@ lx_hd_pipe(intptr_t arg, int flags)
 	 * the pipe capacity is limited to "Fifohiwat" which is a compile-time
 	 * limit set to FIFOHIWAT.
 	 */
+	mutex_enter(&VTOF(vp1)->fn_lock->flk_lock);
 	fifo_fastoff(VTOF(vp1));
+	mutex_exit(&VTOF(vp1)->fn_lock->flk_lock);
 
 	/*
 	 * Set the O_NONBLOCK flag if requested.
